### PR TITLE
Updating documentation to clarify full vs half-bandwidth and defaults in time_frequency.multitaper.py

### LINKED ISSUE #11324: Multitaper keyword "bandwidth" does not behave as expected
### Proposed documentation enhancement

I try to calculate power spectral densities using multitaper at different frequency resolutions but I don't get it to work. The documentation states

```
bandwidth[float]
The bandwidth of the multi taper windowing function in Hz. The default value is a window half-bandwidth of 4.
```

I understand this doc such that 
`mne.time_frequency.psd_multitaper(raw, bandwidth=None)`

should yield the same as 
`mne.time_frequency.psd_multitaper(raw, bandwidth=4)`

but it does not. Looking at the code, I see how `half_nbw` is obtained from `bandwidth`:
https://github.com/mne-tools/mne-python/blob/2d90ecaee807d3d7ceab31bdac0a734bf5418c87/mne/time_frequency/multitaper.py#L341

So I try 
`mne.time_frequency.psd_multitaper(raw, bandwidth=(4 * raw.last_samp) / (2 * raw.info["sfreq"]))`
which again yields something different. 

**My question:**
Which float do I need to pass to `bandwidth` to reproduce `bandwith=None`?

Here is my code:

```Python
import mne
import numpy as np
import matplotlib.pyplot as plt

info = mne.create_info(["eeg1"], sfreq=256, ch_types="eeg")
data = data = np.random.randn(1, 256*10)
raw = mne.io.RawArray(data, info)

bandwidths = [None, 
              4, 
              (4 * raw.last_samp) / (2 * raw.info["sfreq"])]

_, axes = plt.subplots(1, 3, figsize=(10, 5), sharey=True)
for ax, bandwidth in zip(axes, bandwidths):
    raw.compute_psd(method="multitaper", bandwidth=bandwidth).plot(axes=ax)
    ax.set_xlabel("Frequency (Hz)")
    ax.set_title(f"Bandwidth={bandwidth}")
# plt.savefig("bandwidths.png", dpi=150)
plt.show()
```

This is the result:
bandwidths

### PR DESCRIPTION
**Pull request by tomdstone**
#### Reference issue
Fixes #11324

#### What does this implement/fix?
Several folks noticed that the default bandwidth in `time_frequency.psd_multitaper()` was not documented correctly, and that the bandwidth was incorrectly documented as a half-bandwidth. The documentation has been updated to specify that it is the full frequency bandwidth, and the documentation of how the default bandwidth is 'computed'. The old version saying that the default bandwidth was 4 Hz was incorrect, as the 4 was the normalized time-half-bandwidth product.

Also updated the `ValueError` in `_compute_mt_params` when computing `half_nbw` to specify which is full bandwidth and which is half-bandwidth.

### CODE REVIEW

File: mne/time_frequency/multitaper.py [hunk 1]
```diff
@@ -331,9 +331,9 @@ def psd_array_multitaper(x, sfreq, fmin=0.0, fmax=np.inf, bandwidth=None,
         The sampling frequency.
     %(fmin_fmax_psd)s
     bandwidth : float
-        Half-bandwidth of the multi-taper window function in Hz. For a given
-        frequency, frequencies at ± half-bandwidth are smoothed together.
-        The default value is a half-bandwidth of 4.
+        Frequency bandwidth of the multi-taper window function in Hz. For a
+        given frequency, frequencies at ± bandwidth/2 are smoothed together.
+        The default value is a bandwidth of 8 * (sfreq / n_times).
```
Comment (mscheltienne) [inline mne/time_frequency/multitaper.py:336]:
```suggestion
        given frequency, frequencies at ``± bandwidth / 2`` are smoothed together.
        The default value is a bandwidth of ``8 * (sfreq / n_times)``.
```

Thanks for catching this one! I did not look at the code in detail when I edited those docstrings..

File: mne/time_frequency/multitaper.py [hunk 2]
```diff
@@ -294,7 +294,7 @@ def _compute_mt_params(n_times, sfreq, bandwidth, low_bias, adaptive,
         half_nbw = 4.
     if half_nbw < 0.5:
         raise ValueError(
-            'bandwidth value %s yields a normalized bandwidth of %s < 0.5, '
+            'bandwidth value %s yields a normalized half-bandwidth of %s < 0.5, '
             'use a value of at least %s'
```
Comment (larsoner) [inline mne/time_frequency/multitaper.py:298]:
```suggestion
            'bandwidth value %s yields a normalized half-bandwidth of '
            '%s < 0.5, use a value of at least %s'
```

File: mne/time_frequency/multitaper.py [hunk 3]
```diff
@@ -331,9 +331,9 @@ def psd_array_multitaper(x, sfreq, fmin=0.0, fmax=np.inf, bandwidth=None,
         The sampling frequency.
     %(fmin_fmax_psd)s
     bandwidth : float
-        Half-bandwidth of the multi-taper window function in Hz. For a given
-        frequency, frequencies at ± half-bandwidth are smoothed together.
-        The default value is a half-bandwidth of 4.
+        Frequency bandwidth of the multi-taper window function in Hz. For a
+        given frequency, frequencies at ``± bandwidth / 2`` are smoothed together.
+        The default value is a bandwidth of ``8 * (sfreq / n_times)``.
```
Comment (larsoner) [inline mne/time_frequency/multitaper.py:336]:
```suggestion
        given frequency, frequencies at ``± bandwidth / 2`` are smoothed
        together. The default value is a bandwidth of
        ``8 * (sfreq / n_times)``.
```

**Comment by larsoner** [comment]
@britta-wstnr can you look and merge if happy?

**Comment by britta-wstnr** [comment]
@larsoner the failing test is unrelated as far as i can see, right?

**Review by mmagnuski — APPROVED** [review]

**Comment by britta-wstnr** [comment]
Hi @tomdstone
thanks for fixing this, LGTM!
It seems like you are a first time contributor, so please go ahead and add your name to our list of authors `doc/changes/names.inc`.
Could you also please add this change to the change-log under `Bugs`?

You can find a more thorough description about these steps on the MNE Contributor guide https://mne.tools/stable/install/contributing.html under the section "First-time contributors".
Thanks! 🙂

**Comment by drammock** [comment]
@britta-wstnr the failing test should have been fixed by #11478 so I've gone ahead and merged main here, that should fix it. @tomdstone be sure to `git pull` before making the changes to `names.inc` and `docs/changes/latest.inc` that @britta-wstnr requested.

**Comment by tomdstone** [comment]
Changelog and author list updated!

**Pull request merged**